File: src/Sample/Plumix.Sample/Demos/General/SafeAreaDemoPage.cs
using Avalonia;
using Avalonia.Media;
using Plumix.Rendering;
using Plumix.UI;
using Plumix.Widgets;

namespace Plumix;

// Dart parity source: dart_sample/lib/demos/general/safe_area_demo_page.dart (exact sample parity)

public sealed class SafeAreaDemoPage : StatelessWidget
{
    public override Widget Build(BuildContext context)
    {
        return new Column(
            crossAxisAlignment: CrossAxisAlignment.Stretch,
            spacing: 12,
            children:
            [
                new Text("SafeArea", fontSize: 20, color: Colors.Black),
                new Text(
                    "The rose surface is the simulated system intrusion. The blue child keeps a minimum " +
                    "8 px inset and preserves the 28 px bottom view padding consumed by a keyboard.",
                    fontSize: 14,
                    color: Colors.DimGray),
                BuildBoxPreview(),
                new Text(
                    "SliverSafeArea applies the same edge policy in sliver geometry.",
                    fontSize: 14,
                    color: Colors.DimGray),
                new Expanded(child: BuildSliverPreview()),
            ]);
    }

    private static Widget BuildBoxPreview()
    {
        return new MediaQuery(
            data: new MediaQueryData(
                Padding: new Thickness(24, 18, 32, 0),
                ViewPadding: new Thickness(24, 18, 32, 28)),
            child: new Container(
                height: 170,
                color: Color.Parse("#FFFFCDD2"),
                child: new SafeArea(
                    minimum: EdgeInsets.All(8),
                    maintainBottomViewPadding: true,
                    child: new Container(
                        alignment: Alignment.Center,
                        color: Color.Parse("#FFBBDEFB"),
                        child: new Text(
                            "Safe content\n24 left · 18 top · 32 right · 28 bottom",
                            fontSize: 15,
                            color: Colors.Black,
                            textAlign: TextAlign.Center)))));
    }

    private static Widget BuildSliverPreview()
    {
        return new MediaQuery(
            data: new MediaQueryData(
                Padding: new Thickness(16, 20, 24, 12),
                ViewPadding: new Thickness(16, 20, 24, 12)),
            child: new Container(
                color: Color.Parse("#FFFFE0E0"),
                child: new CustomScrollView(
                    slivers:
                    [
                        new SliverSafeArea(
                            minimum: EdgeInsets.All(8),
                            sliver: SliverFixedExtentList.Builder(
                                childCount: 8,
                                itemExtent: 44,
                                itemBuilder: (_, index) => new Container(
                                    alignment: Alignment.CenterLeft,
                                    color: index % 2 == 0
                                        ? Color.Parse("#FFE3F2FD")
                                        : Color.Parse("#FFFFFFFF"),
                                    padding: new Thickness(12, 8),
                                    child: new Text(
                                        $"safe sliver row #{index}",
                                        fontSize: 14,
                                        color: Colors.Black)),
                                addAutomaticKeepAlives: false)),
                    ])));
    }
}

File: dart_sample/lib/demos/general/safe_area_demo_page.dart
import 'package:material_ui/material_ui.dart';

/// C# parity source: src/Sample/Plumix.Sample/Demos/General/SafeAreaDemoPage.cs
class SafeAreaDemoPage extends StatelessWidget {
  const SafeAreaDemoPage({super.key});

  @override
  Widget build(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.stretch,
      spacing: 12,
      children: <Widget>[
        const Text(
          'SafeArea',
          style: TextStyle(fontSize: 20, color: Colors.black),
        ),
        const Text(
          'The rose surface is the simulated system intrusion. The blue child '
          'keeps a minimum 8 px inset and preserves the 28 px bottom view '
          'padding consumed by a keyboard.',
          style: TextStyle(fontSize: 14, color: Colors.black54),
        ),
        _buildBoxPreview(),
        const Text(
          'SliverSafeArea applies the same edge policy in sliver geometry.',
          style: TextStyle(fontSize: 14, color: Colors.black54),
        ),
        Expanded(child: _buildSliverPreview()),
      ],
    );
  }

  static Widget _buildBoxPreview() {
    return MediaQuery(
      data: const MediaQueryData(
        padding: EdgeInsets.fromLTRB(24, 18, 32, 0),
        viewPadding: EdgeInsets.fromLTRB(24, 18, 32, 28),
      ),
      child: SizedBox(
        height: 170,
        child: ColoredBox(
          color: const Color(0xFFFFCDD2),
          child: SafeArea(
            minimum: const EdgeInsets.all(8),
            maintainBottomViewPadding: true,
            child: ColoredBox(
              color: const Color(0xFFBBDEFB),
              child: Center(
                child: Text(
                  'Safe content\n24 left · 18 top · 32 right · 28 bottom',
                  textAlign: TextAlign.center,
                  style: TextStyle(fontSize: 15, color: Colors.black),
                ),
              ),
            ),
          ),
        ),
      ),
    );
  }

  static Widget _buildSliverPreview() {
    return MediaQuery(
      data: const MediaQueryData(
        padding: EdgeInsets.fromLTRB(16, 20, 24, 12),
        viewPadding: EdgeInsets.fromLTRB(16, 20, 24, 12),
      ),
      child: ColoredBox(
        color: const Color(0xFFFFE0E0),
        child: CustomScrollView(
          slivers: <Widget>[
            SliverSafeArea(
              minimum: const EdgeInsets.all(8),
              sliver: SliverFixedExtentList.builder(
                itemCount: 8,
                itemExtent: 44,
                itemBuilder: (BuildContext context, int index) {
                  return ColoredBox(
                    color: index.isEven
                        ? const Color(0xFFE3F2FD)
                        : const Color(0xFFFFFFFF),
                    child: Padding(
                      padding: const EdgeInsets.symmetric(
                        horizontal: 12,
                        vertical: 8,
                      ),
                      child: Align(
                        alignment: Alignment.centerLeft,
                        child: Text(
                          'safe sliver row #$index',
                          style: const TextStyle(
                            fontSize: 14,
                            color: Colors.black,
                          ),
                        ),
                      ),
                    ),
                  );
                },
              ),
            ),
          ],
        ),
      ),
    );
  }
}
